assets/js/src/segments/dynamic/form.tsx

```diff
@@ -85,6 +86,18 @@ const DynamicSegmentForm: React.FunctionComponent = () => {
       return found;
     }
 
+    function convertSavedData(data: {
+      [key: string]: string | number;
+    }): AnyFormItem {
+      const converted = JSON.parse(JSON.stringify(data));
+      // for compatibility with older data
+      if (has('link_id', data)) converted.link_id = data.link_id.toString();
+      if (has('newsletter_id', data)) converted.newsletter_id = data.newsletter_id.toString();
+      if (has('product_id', data)) converted.product_id = data.product_id.toString();
+      if (has('category_id', data)) converted.category_id = data.category_id.toString();
+      return converted;
+    }
+
```

Comment:
This doesn't fix that some data that used to be stored as integers would be stored as strings. I'm afraid that this may cause some unexpected issues. 
For example, I wanted to test the email link clicked segment but for some reason, I'm not able to create or edit such a segment, because I can't select and save the link. I recorded video https://d.pr/v/qL20X9. I don't get any error in the console but the input just disappears and appears again without the selected link.

Maybe we could fix it the other way around by altering components for selecting emails, links, products, and categories to accept integers. What do you think?

Reply:
> Maybe we could fix it the other way around by altering components for selecting emails, links, products, and categories to accept integers. What do you think?

Yes, that would be better, but I am afraid that is not possible because we use a library that doesn't support that.